### ISSUE TITLE
[BUG] Group Membership - Rejected vs Denied and display discrepancy

### ISSUE THREAD

**Issue by kpecknold** [rokwire/illinois-app#482]
1. When a requestor is denied by an admin, the status is currently "Rejected". This should be changed to "Denied" to be consistent with the language on the admin side (and the term "Denied" is a little less offensive)
2. When a requestor is waiting, their status on the groups home screen is "Pending". Once they've been denied, the status disappears completely from the groups home screen and can only been seen when tapping into a group.

**To Reproduce**
Requires a 2 user test
1. User requests to join a group, observes status on groups home page and group detail page (refreshes don't happen unless going in and out?)
2. Admin of that group denies the request
3. User again observes both the groups home and the detail to see the discrepancy in the display of status

**Expected behavior**
Status display should be consistent

**Comment by vburgett** [comment]
This ticket is related to ticket #5098 in that we would like the word "rejected" to now be "denied." cc @mihail-varbanov

**Comment by mihail-varbanov** [comment]
@vburgett, I do not see relation of this issue with #5098. As far as I understand, #5098 deals with new user selection using the directory of users, that affects only regular & admin users. This issue is about the term that we use to refer to denied/rejected users. For historical reasons, we use the "rejected" keyword as an internal status name, but we take care to use "denied" when referring to this status everywhere in the UI.

**Comment by vburgett** [comment]
Understood, @mihail-varbanov. "Rejected" as an internal reference is fine. The UI does not yet reflect the request for updating "Rejected" to "Denied."

![Image](https://github.com/user-attachments/assets/ffd365ae-aa8c-44eb-b18e-44d8695464bc)

**Comment by mihail-varbanov** [comment]
Yes indeed, @vburgett. We should use "denied" instead of "rejected" everywhere after versions 7.0.21 & 7.1.4.

**Comment by vburgett** [comment]
The denial notification should also replace "rejected" using this text:

Group "test group 73"
Your membership request has been denied. The reason provided was: "test"

![Screenshot_20250520-151441~2.png](https://github.com/user-attachments/assets/0777981f-20f5-44b0-98ac-f7111399daae)

**Comment by mihail-varbanov** [comment]
@vburgett, this text is generated on the backend side. I created an issue for this in the groups BB repo: [#574](https://github.com/rokwire/groups-building-block/issues/574).

**Comment by mdryankov** [comment]
Fixed on groups bb side

**Comment by vburgett** [comment]
@mihail-varbanov all looks complete now on dev in app version 7.1.10. ![image](https://github.com/user-attachments/assets/9240a35d-9c21-4151-bcd7-4e44ae124c9a)
